__tests__/app.test.js

```diff
@@ -1,280 +1,346 @@
-    })
-    
-})
+		test("responds with a 400 status when given an incomplete request", () => {
+			const testComment = {
+				username: "rogersop",
+			};
+			return request(app)
+				.post("/api/articles/3/comments")
+				.send(testComment)
+				.expect(400)
+				.then((response) => {
+					expect(response.body.msg).toEqual("Bad request");
+				});
+		});
 
+		test("responds with a 400 status when given an invalid data type", () => {
+			const testComment = {
+				username: 1234,
+				body: "Wow. I, for one, am shocked and appalled!!!",
+			};
+			return request(app)
+				.post("/api/articles/3/comments")
+				.send(testComment)
+				.expect(400)
+				.then((response) => {
+					expect(response.body.msg).toEqual("Bad request");
+				});
+		});
+	});
+});
 
+//// question 8 patchArticleVotes
 
+describe("PATCH /api/articles/:article_id", () => {
+	describe("behaviours", () => {
+		test("PATCH:200 takes a number argument and updates the vote value on an article given the article id ", () => {
+			const updatedVotes = {
+				inc_votes: 6,
+			};
+			return request(app)
+				.patch("/api/articles/1")
+				.send(updatedVotes)
+				.expect(200)
+				.then((response) => {
+					expect(response.body.vote.votes).toBe(106);
+				});
+		});
+		test("PATCH:200 takes a negative number argument and updates the vote value on an article given the article id ", () => {
+			const updatedVotes = {
+				inc_votes: -17,
+			};
+			return request(app)
+				.patch("/api/articles/1")
+				.send(updatedVotes)
+				.expect(200)
+				.then((response) => {
+					expect(response.body.vote.votes).toBe(83);
+				});
+		});
+		test("PATCH:200 takes an argument with no value and the vote value stays the same", () => {
+			const updatedVotes = {
+				inc_votes: 0,
+			};
+			return request(app)
+				.patch("/api/articles/1")
+				.send(updatedVotes)
+				.expect(200)
+				.then((response) => {
+					expect(response.body.vote.votes).toBe(100);
+				});
+		});
 
+		test("works when the article does not currently have any votes to display", () => {
+			const updatedVotes = {
+				inc_votes: 10,
+			};
+			return request(app)
+				.patch("/api/articles/2")
+				.send(updatedVotes)
+				.expect(200)
+                .then((response) => {
+					expect(response.body.vote.votes).toBe(10);
+				});
+		});
+	});
+	describe("Error handling", () => {
+		test("takes an empty object and returns 400", () => {
+			const updatedVotes = {};
+			return request(app)
+				.patch("/api/articles/1")
+				.send(updatedVotes)
+                .expect(400)
+                .then((response) => {
+					expect(response.body.msg).toEqual("Bad request");
+				});
+		});
 
+		test("takes an object with null value and returns 400", () => {
+			const updatedVotes = {
+				inc_votes: null,
+			};
+			return request(app)
+				.patch("/api/articles/1")
+				.send(updatedVotes)
+				.expect(400)
+				.then((response) => {
+					expect(response.body.msg).toEqual("Bad request");
+				});
+            
+		});
+	});
```

Comment:
2 tests missing here
Status 400, invalid ID, e.g. string of "not-an-id"
Status 404, non existent ID, e.g. 0 or 9999